hashicorp/tfc-workflows-tooling
# feat: allow global --json flag usage

## Description

Add support for other command line tools such as `jq` by allowing a global `--json` flag argument.

Currently, we are outputting some in-progress/diagnostic information to stdout rather than to stderr. When a `--json` flag is detected regardless of position, we will send all information that is not the final result set to the standard error stream.

A new [writer](https://github.com/hashicorp/tfc-workflows-tooling/blob/0c61894f699825c228be23f6bd74e3dbd4f6cd38/internal/writer/writer.go#L12) package has been added to allow for better dependency injection for other packages.

## Testing plan

1. Execute various commands **without** the use of `--json` flag. No change in behavior and output.
1. Run and verify the following script works as expected, where the apply succeeds.

<details>
  <summary>script.sh</summary>

```bash
# set directory and workspace
DIR=""
WORKSPACE=""

CV_ID=$(go run . upload \
  --workspace=$WORKSPACE \
  --directory=$DIR \
  --json \
  | jq -r '.configuration_version_id'
)

echo $CV_ID

RUN_ID=$(go run . run create \
  --workspace=$WORKSPACE \
  --configuration_version=$CV_ID \
  --json \
  | jq -r '.run_id'
)

RES=$(
  go run . run apply \
  --run=$RUN_ID \
  --json  \
)

echo $RES
```

</details>


## External links
